packages/backend-modules/mail/MailchimpInterface.js

```diff
@@ -14,16 +14,24 @@ const MINIMUM_HTTP_RESPONSE_STATUS_ERROR = 400
 const MailchimpInterface = ({ logger }) => {
   checkEnv(['MAILCHIMP_API_KEY', 'MAILCHIMP_URL', 'MAILCHIMP_MAIN_LIST_ID'])
   return {
-    buildApiUrl(path) {
-      return `${MAILCHIMP_URL}/3.0/lists/${MAILCHIMP_MAIN_LIST_ID}${path}`
+    buildApiUrl(audienceId, path) {
+      return `${MAILCHIMP_URL}/3.0/lists/${audienceId}${path}`
+    },
+    buildAudienceApiUrl(email, audienceId) {
+      const hash = crypto
+        .createHash('md5')
+        .update(email.toLowerCase())
+        .digest('hex')
+
+      return this.buildApiUrl(audienceId, `/members/${hash}`)
     },
     buildMembersApiUrl(email) {
```

Comment:
```suggestion
    buildMembersApiUrl(email, audienceId = MAILCHIMP_MAIN_LIST_ID) {
```
if audienceId is undefined the function would use the given default, i.e. MAILCHIMP_MAIN_LIST_ID

Reply:
then you could use `buildMembersApiUrl` everywhere and delete `buildAudienceApiUrl`